src/Domain/Services/DraftOrders.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * Sets up all logic related to the Checkout Draft Orders service
+ *
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\Domain\Services;
+
+use Automattic\WooCommerce\Blocks\Domain\Package;
+
+/**
+ * Service class for adding DraftOrder functionality to WooCommerce core.
+ */
+class DraftOrders {
+
+	/**
+	 * Holds the Package instance
+	 *
+	 * @var Package
+	 */
+	private $package;
+
+	/**
+	 * Constructor
+	 *
+	 * @param Package $package An instance of the package class.
+	 */
+	public function __construct( Package $package ) {
+		$this->package = $package;
+	}
+
+	/**
+	 * Set all hooks related to adding Checkout Draft order functionality to Woo Core.
+	 */
+	public function init() {
+		if ( $this->package->is_feature_plugin_build() ) {
+			add_filter( 'wc_order_statuses', [ __CLASS__, 'register_draft_order_status' ] );
+			add_filter( 'woocommerce_register_shop_order_post_statuses', [ __CLASS__, 'register_draft_order_post_status' ] );
+			add_filter( 'woocommerce_valid_order_statuses_for_payment', [ __CLASS__, 'append_draft_order_post_status' ] );
+			add_action( 'woocommerce_cleanup_draft_orders', [ __CLASS__, 'delete_expired_draft_orders' ] );
+			add_action( 'admin_init', [ $this, 'install' ] );
+		} else {
+			// Maybe remove existing cronjob if present because it shouldn't be needed in the environment.
+			$this->maybe_remove_cronjobs();
```

Comment:
Any reason why install waits for admin_init and this one does not?

Reply:
hmm ya it probably should wait for admin_init too.